server/mailet/rate-limiter/src/main/scala/org/apache/james/transport/mailets/PerSenderRateLimit.scala

```diff
@@ -147,55 +109,26 @@ class PerSenderRateLimit @Inject()(rateLimiterFactoryProvider: RateLimiterFactor
     keyPrefix = Option(getInitParameter("keyPrefix")).map(KeyPrefix)
     exceededProcessor = getInitParameter("exceededProcessor", Mail.ERROR)
 
-    countRateLimiter = PerSenderRateLimiter(rateLimiter = extractCountRules(duration).map(rateLimiterFactory.withSpecification),
+    countRateLimiter = PerSenderRateLimiter(rateLimiter = Count.extractRules(duration, getInitParameter("count"))
+      .map(rateLimiterFactory.withSpecification),
       keyPrefix = keyPrefix,
       entityType = Count)
-    recipientsRateLimiter = PerSenderRateLimiter(rateLimiter = extractRecipientsRules(duration).map(rateLimiterFactory.withSpecification),
+    recipientsRateLimiter = PerSenderRateLimiter(rateLimiter = RecipientsType.extractRules(duration, getInitParameter("recipients"))
+      .map(rateLimiterFactory.withSpecification),
       keyPrefix = keyPrefix,
       entityType = RecipientsType)
-    sizeRateLimiter = PerSenderRateLimiter(rateLimiter = extractSizeRules(duration).map(rateLimiterFactory.withSpecification),
+    sizeRateLimiter = PerSenderRateLimiter(rateLimiter = Size.extractRules(duration, getInitParameter("size"))
+      .map(rateLimiterFactory.withSpecification),
       keyPrefix = keyPrefix,
       entityType = Size)
-    totalSizeRateLimiter = PerSenderRateLimiter(rateLimiter = extractTotalSizeRules(duration).map(rateLimiterFactory.withSpecification),
+    totalSizeRateLimiter = PerSenderRateLimiter(rateLimiter = TotalSize.extractRules(duration, getInitParameter("totalSize"))
+      .map(rateLimiterFactory.withSpecification),
       keyPrefix = keyPrefix,
       entityType = TotalSize)
```

Comment:
Given that we pass the MailetConfig to the entityType we could further factorize this:

```
private def createRateLimiter(rateLimiterFactory: RateLimiterFactory, entityType: EntityType, keyPrefix: KeyPrefix, duration: Duration): PerSenderRateLimiter = 
    PerSenderRateLimiter(rateLimiter = entityType.extractRules(duration, getMailetConfig())
          .map(rateLimiterFactory.withSpecification),
          keyPrefix = keyPrefix,
          entityType = entityType)
// ...
def perSenderRateLimiter(entityType: EntityType): PerSenderRateLimiter = createRateLimiter(rateLimiterFactory, entityType, keyPrefix, duration) 
countRateLimiter = perSenderRateLimiter(Count)
recipientsRateLimiter = perSenderRateLimiter(Recipients)
sizeRateLimiter = perSenderRateLimiter(Size)
totalSizeRateLimiter = perSenderRateLimiter(TotalSize)
```

Here and in the other mailet too. More beautiful isn't it? 

(notice the use of a local method to limit the count of arguments passed. Implicits would work too...